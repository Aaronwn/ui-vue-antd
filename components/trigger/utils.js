function isPointsEq (a1, a2) {
  return a1[0] === a2[0] && a1[1] === a2[1]
}

export function getAlignFromPlacement (builtinPlacements, placementStr, align) {
  const baseAlign = builtinPlacements[placementStr] || {}
  return {
    ...baseAlign,
    ...align,
  }
}

export function getPopupClassNameFromAlign (builtinPlacements, prefixCls, align) {
  const points = align.points
  for (const placement in builtinPlacements) {
    if (builtinPlacements.hasOwnProperty(placement)) {
      if (isPointsEq(builtinPlacements[placement].points, points)) {
        return `${prefixCls}-placement-${placement}`
      }
    }
  }
  return ''
}
export function noop () {
}
