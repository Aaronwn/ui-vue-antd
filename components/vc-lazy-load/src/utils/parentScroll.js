const style = (element, prop) => {
  let styleVal = ''
  if (typeof getComputedStyle !== 'undefined') {
    styleVal = window.getComputedStyle(element, null).getPropertyValue(prop)
  } else {
    styleVal = element.style[prop]
  }
  return styleVal
}

const overflow = (element) =>
  style(element, 'overflow') + style(element, 'overflow-y') + style(element, 'overflow-x')

const scrollParent = (element) => {
  if (!(element instanceof window.HTMLElement)) {
    return window
  }

  let parent = element

  while (parent) {
    if (parent === document.body || parent === document.documentElement) {
      break
    }

    if (!parent.parentNode) {
      break
    }
    if (/(scroll|auto)/.test(overflow(parent))) {
      return parent
    }

    parent = parent.parentNode
  }

  return window
}

export default scrollParent
