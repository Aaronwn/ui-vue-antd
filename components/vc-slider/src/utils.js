import keyCode from '../../_util/KeyCode'

export function isEventFromHandle (e, handles) {
  console.log(Object.keys(handles)
    .some(key => e.target === handles[key].$el))
  return Object.keys(handles)
    .some(key => e.target === handles[key].$el)
}

export function isValueOutOfRange (value, { min, max }) {
  return value < min || value > max
}

export function isNotTouchEvent (e) {
  return e.touches.length > 1 ||
    (e.type.toLowerCase() === 'touchend' && e.touches.length > 0)
}

export function getClosestPoint (val, { marks, step, min }) {
  const points = Object.keys(marks).map(parseFloat)
  if (step !== null) {
    const closestStep =
            Math.round((val - min) / step) * step + min
    points.push(closestStep)
  }
  const diffs = points.map(point => Math.abs(val - point))
  return points[diffs.indexOf(Math.min(...diffs))]
}

export function getPrecision (step) {
  const stepString = step.toString()
  let precision = 0
  if (stepString.indexOf('.') >= 0) {
    precision = stepString.length - stepString.indexOf('.') - 1
  }
  return precision
}

export function getMousePosition (vertical, e) {
  return vertical ? e.clientY : e.pageX
}

export function getTouchPosition (vertical, e) {
  return vertical ? e.touches[0].clientY : e.touches[0].pageX
}

export function getHandleCenterPosition (vertical, handle) {
  const coords = handle.getBoundingClientRect()
  return vertical
    ? coords.top + (coords.height * 0.5)
    : coords.left + (coords.width * 0.5)
}

export function ensureValueInRange (val, { max, min }) {
  if (val <= min) {
    return min
  }
  if (val >= max) {
    return max
  }
  return val
}

export function ensureValuePrecision (val, props) {
  const { step } = props
  const closestPoint = getClosestPoint(val, props)
  return step === null ? closestPoint
    : parseFloat(closestPoint.toFixed(getPrecision(step)))
}

export function pauseEvent (e) {
  e.stopPropagation()
  e.preventDefault()
}

export function getKeyboardValueMutator (e) {
  switch (e.keyCode) {
    case keyCode.UP:
    case keyCode.RIGHT:
      return (value, props) => value + props.step

    case keyCode.DOWN:
    case keyCode.LEFT:
      return (value, props) => value - props.step

    case keyCode.END: return (value, props) => props.max
    case keyCode.HOME: return (value, props) => props.min
    case keyCode.PAGE_UP: return (value, props) => value + props.step * 2
    case keyCode.PAGE_DOWN: return (value, props) => value - props.step * 2

    default: return undefined
  }
}

export function getComponentProps (obj, prop) {
  if (obj[prop]) {
    return obj
  } else if (obj.$children.length) {
    const len = obj.$children.length
    for (let i = 0; i < len; i++) {
      if (obj.$children[i][prop]) {
        return obj.$children[i]
      } else if (obj.$children[i].$children.length) {
        return getComponentProps(obj.$children[i], prop)
      }
    }
  }
}
