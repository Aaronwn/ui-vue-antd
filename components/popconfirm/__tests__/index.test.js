import { mount } from '@vue/test-utils'
import { asyncExpect } from '@/tests/utils'
import Popconfirm from '..'
function $$ (className) {
  return document.body.querySelectorAll(className)
}
describe('Popconfirm', () => {
  it('should popup Popconfirm dialog', async () => {
    const onVisibleChange = jest.fn()

    const wrapper = mount(
      {
        render () {
          return <Popconfirm
            title={<span class='popconfirm-test'>Are you sure delete this task?</span>}
            okText='Yes'
            cancelText='No'
            mouseEnterDelay={0}
            mouseLeaveDelay={0}
            onVisibleChange={onVisibleChange}
          >
            <span>Delete</span>
          </Popconfirm>
        },
      }, { sync: false, attachToDocument: true })
    let triggerNode = null
    await asyncExpect(() => {
      triggerNode = wrapper.findAll('span').at(0)
      triggerNode.trigger('click')
    })
    await asyncExpect(() => {
      expect(onVisibleChange).toBeCalledWith(true)
      expect($$('.popconfirm-test').length).toBe(1)
      triggerNode.trigger('click')
    }, 1000)
    await asyncExpect(() => {
      expect(onVisibleChange).toBeCalledWith(false)
    })
  })

  it('should show overlay when trigger is clicked', async () => {
    const popconfirm = mount({
      render () {
        return <Popconfirm ref='popconfirm' title='code'>
          <span>show me your code</span>
        </Popconfirm>
      },
    }, { sync: false })

    await asyncExpect(() => {
      expect(popconfirm.vm.$refs.popconfirm.getPopupDomNode()).toBe(null)

      popconfirm.find('span').trigger('click')
    }, 1000)
    await asyncExpect(() => {
      const popup = popconfirm.vm.$refs.popconfirm.getPopupDomNode()
      expect(popup).not.toBe(null)
      expect(popup.className).toContain('ant-popover-placement-top')
      expect(popup.innerHTML).toMatchSnapshot()
    }, 1000)
  })
})
