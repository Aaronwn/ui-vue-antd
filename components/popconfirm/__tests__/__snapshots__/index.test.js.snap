// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popconfirm should show overlay when trigger is clicked 1`] = `
<div class="ant-popover-content">
  <div class="ant-popover-arrow"></div>
  <div class="ant-popover-inner">
    <div class="ant-popover-inner-content">
      <div class="ant-popover-message">
        <i class="anticon anticon-exclamation-circle"></i>
        <div class="ant-popover-message-title">code</div>
      </div>
      <div class="ant-popover-buttons">
        <button type="button" class="ant-btn ant-btn-default ant-btn-sm"><span>Cancel</span></button>
        <button type="button" class="ant-btn ant-btn-primary ant-btn-sm"><span>OK</span></button>
      </div>
    </div>
  </div>
</div>
`;
